dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/CoreFeatureTest.kt

```diff
@@ -464,6 +473,44 @@ internal class CoreFeatureTest {
         assertThat(testedFeature.uploadFrequency).isEqualTo(fakeConfig.coreConfig.uploadFrequency)
     }
 
+    @Test
+    fun `𝕄 initializes build ID 𝕎 initialize()`() {
+        // When
+        testedFeature.initialize(
+            appContext.mockInstance,
+            fakeSdkInstanceId,
+            fakeConfig,
+            fakeConsent
+        )
+
+        // Then
+        assertThat(testedFeature.appBuildId).isEqualTo(fakeBuildId.toString())
+    }
+
+    @Test
+    fun `𝕄 initializes build ID 𝕎 initialize() { build ID is missing }`() {
+        // Given
+        whenever(
+            appContext.mockInstance.assets.open(CoreFeature.BUILD_ID_FILE_NAME)
+        ) doThrow IOException()
```

Comment:
> [!NOTE]
> This should be a `FileNotFoundException` to properly mock an asset not being present.

Reply:
You can add other tests for other scenarios: 
- `RuntimeException` could be thrown if the `AssetManager` has been closed (cf: `AssetManager.ensureOpenLocked()`
- `IOException` if the `read()` operation fails